pkg/management/extensions.go

```diff
@@ -316,7 +316,44 @@ func newHandler(
 				status.Errorf(codes.InvalidArgument, err.Error()))
 			return
 		} else if len(body) > 0 {
-			if err := reqMsg.UnmarshalMergeJSON(body); err != nil {
+			var err error
+			bodyFieldPath := rule.GetHttp().GetBody()
+			if bodyFieldPath == "" {
+				bodyFieldPath = "*"
+			}
+			if bodyFieldPath == "*" {
+				err = reqMsg.UnmarshalMergeJSON(body)
+			} else {
+				// unmarshal the request into a specific field
+				fd := reqMsg.FindFieldDescriptorByJSONName(bodyFieldPath)
+				if fd == nil {
+					runtime.HTTPError(ctx, mux, outboundMarshaler, w, req,
+						status.Errorf(codes.InvalidArgument, "unknown field %q", bodyFieldPath))
+					return
+				}
+				if fd.GetType() == dpb.FieldDescriptorProto_TYPE_MESSAGE {
+					bodyField := reqMsg.GetField(fd)
+					var dynamicBodyField *dynamic.Message
+					if reflect.ValueOf(bodyField).IsNil() {
+						dynamicBodyField = dynamic.NewMessage(fd.GetMessageType())
+						reqMsg.SetField(fd, dynamicBodyField)
+					} else {
+						var ok bool
+						dynamicBodyField, ok = bodyField.(*dynamic.Message)
+						if !ok {
+							panic("bug: request is not a *dynamic.Message")
```

Comment:
I can see it both ways but returning `InternalServerError` here might make sense here

Reply:
good catch, this should be handled differently

Reply:
actually this looks correct, upon further inspection the dynamic package will always unmarshal by creating nested dynamic messages.

Reply:
yeah i agree this control flow is clearly unrecoverable, just a question around considering error semantics